csharp/Adapter/Microsoft.Spark.CSharp/Services/LoggerServiceFactory.cs

```diff
@@ -11,17 +12,15 @@ namespace Microsoft.Spark.CSharp.Services
     /// </summary>
     public class LoggerServiceFactory
     {
-        private static ILoggerService loggerService = DefaultLoggerService.Instance;
-        
+        private static Lazy<ILoggerService> loggerService = new Lazy<ILoggerService>(() => GetDefaultLogger());
+
         /// <summary>
         /// Overrides an existing logger by a given logger service instance
         /// </summary>
         /// <param name="loggerServiceOverride">The logger service instance used to overrides</param>
         public static void SetLoggerService(ILoggerService loggerServiceOverride)
```

Comment:
### It's compatible:
#### (1) If xxx.exe.config exists and has log4net nodes : Fine :
```
d:\msgit\lzMobius\examples\Batch\WordCount\bin\Release>%SPARKCLR_HOME%\scripts\sparkclr-submit.cmd --exe SparkClrWordCount.exe %cd% SparkClrWordCount.exe.config
SPARKCLR_JAR=spark-clr_2.10-1.6.200-SNAPSHOT.jar
[sparkclr-submit.cmd] Command to run --name SparkClrWordCount --class org.apache.spark.deploy.csharp.CSharpRunner D:\msgit\lzMobius\build\runtime\lib\spark-clr_2.10-1.6.200-SNAPSHOT.jar d:\msgit\lzMobius\examples\Batch\WordCount\bin\Release d:\msgit\lzMobius\examples\Batch\WordCount\bin\Release\SparkClrWordCount.exe  SparkClrWordCount.exe.config
[CSharpRunner.main] Starting CSharpBackend!
[CSharpRunner.main] Port number used by CSharpBackend is 33821
[CSharpRunner.main] adding key=spark.jars and value=file:/D:/msgit/lzMobius/build/runtime/lib/spark-clr_2.10-1.6.200-SNAPSHOT.jar to environment
[CSharpRunner.main] adding key=spark.app.name and value=SparkClrWordCount to environment
[CSharpRunner.main] adding key=spark.submit.deployMode and value=client to environment
[CSharpRunner.main] adding key=spark.master and value=local[*] to environment
[2016-07-14 08:45:02,978] [1] [INFO ] [Microsoft.Spark.CSharp.Configuration.ConfigurationService] - ConfigurationService runMode is LOCAL
[2016-07-14 08:45:02,983] [1] [INFO ] [Microsoft.Spark.CSharp.Configuration.ConfigurationService+SparkCLRConfiguration] - CSharpBackend successfully read from environment variable CSHARPBACKEND_PORT
[2016-07-14 08:45:02,985] [1] [INFO ] [Microsoft.Spark.CSharp.Proxy.Ipc.SparkCLRIpcProxy] - CSharpBackend port number to be used in JvMBridge is 33821
[2016-07-14 08:45:03,050] [3] [DEBUG] [Microsoft.Spark.CSharp.Interop.Ipc.WeakObjectManagerImpl] - Checking objects thread start ...

```
#### (2) If xxx.exe.config exists but no log4net nodes: log4net will show a WARNING : 
"log4net:ERROR Failed to find configuration section 'log4net' " :
```
d:\msgit\lzMobius\examples\Batch\WordCount\bin\Release>%SPARKCLR_HOME%\scripts\sparkclr-submit.cmd --exe SparkClrWordCount.exe %cd% SparkClrWordCount.exe.config
SPARKCLR_JAR=spark-clr_2.10-1.6.200-SNAPSHOT.jar
[sparkclr-submit.cmd] Command to run --name SparkClrWordCount --class org.apache.spark.deploy.csharp.CSharpRunner D:\msgit\lzMobius\build\runtime\lib\spark-clr_2.10-1.6.200-SNAPSHOT.jar d:\msgit\lzMobius\examples\Batch\WordCount\bin\Release d:\msgit\lzMobius\examples\Batch\WordCount\bin\Release\SparkClrWordCount.exe  SparkClrWordCount.exe.config
[CSharpRunner.main] Starting CSharpBackend!
[CSharpRunner.main] Port number used by CSharpBackend is 33572
[CSharpRunner.main] adding key=spark.jars and value=file:/D:/msgit/lzMobius/build/runtime/lib/spark-clr_2.10-1.6.200-SNAPSHOT.jar to environment
[CSharpRunner.main] adding key=spark.app.name and value=SparkClrWordCount to environment
[CSharpRunner.main] adding key=spark.submit.deployMode and value=client to environment
[CSharpRunner.main] adding key=spark.master and value=local[*] to environment
log4net:ERROR Failed to find configuration section 'log4net' in the application's .config file. Check your .config file for the <log4net> and <configSections> elements. The configuration section should look like: <section name="log4net" type="log4net.Config.Log4NetConfigurationSectionHandler,log4net" />
input args: [-m pyspark.worker]
[2016-07-14T00:35:58.3942958Z] [QUAPC] [Info] [Worker] RunSimpleWorker ...
[2016-07-14T00:35:58.3952964Z] [QUAPC] [Debug] [Worker] Files available in executor
[2016-07-14T00:35:58.3952964Z] [QUAPC] [Debug] [Worker] d:\msgit\lzMobius\examples\Batch\WordCount\bin\Release\CSharpWorker.exe
[2016-07-14T00:35:58.3952964Z] [QUAPC] [Debug] [Worker] d:\msgit\lzMobius\examples\Batch\WordCount\bin\Release\log4net.dll
```
#### (3) If xxx.exe.config not exists :  Same as old : 
```
d:\msgit\lzMobius\examples\Batch\WordCount\bin\Release>%SPARKCLR_HOME%\scripts\sparkclr-submit.cmd --exe SparkClrWordCount.exe %cd% log4net.xml
SPARKCLR_JAR=spark-clr_2.10-1.6.200-SNAPSHOT.jar
[sparkclr-submit.cmd] Command to run --name SparkClrWordCount --class org.apache.spark.deploy.csharp.CSharpRunner D:\msgit\lzMobius\build\runtime\lib\spark-clr_2.10-1.6.200-SNAPSHOT.jar d:\msgit\lzMobius\examples\Batch\WordCount\bin\Release d:\msgit\lzMobius\examples\Batch\WordCount\bin\Release\SparkClrWordCount.exe  log4net.xml
[CSharpRunner.main] Starting CSharpBackend!
[CSharpRunner.main] Port number used by CSharpBackend is 3728
[CSharpRunner.main] adding key=spark.jars and value=file:/D:/msgit/lzMobius/build/runtime/lib/spark-clr_2.10-1.6.200-SNAPSHOT.jar to environment
[CSharpRunner.main] adding key=spark.app.name and value=SparkClrWordCount to environment
[CSharpRunner.main] adding key=spark.submit.deployMode and value=client to environment
[CSharpRunner.main] adding key=spark.master and value=local[*] to environment
log4net:ERROR Failed to find configuration section 'log4net' in the application's .config file. Check your .config file for the <log4net> and <configSections> elements. The configuration section should look like: <section name="log4net" type="log4net.Config.Log4NetConfigurationSectionHandler,log4net" />
input args: [-m pyspark.worker]
[2016-07-14T01:30:55.1480914Z] [QUAPC] [Info] [Worker] RunSimpleWorker ...
[2016-07-14T01:30:55.1485918Z] [QUAPC] [Debug] [Worker] Files available in executor
[2016-07-14T01:30:55.1490922Z] [QUAPC] [Debug] [Worker] d:\msgit\lzMobius\examples\Batch\WordCount\bin\Release\CSharpWorker.exe
[2016-07-14T01:30:55.1490922Z] [QUAPC] [Debug] [Worker] d:\msgit\lzMobius\examples\Batch\WordCount\bin\Release\log4net.dll

```